src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -1084,15 +1072,16 @@ void ProtocolHandlerImpl::OnTMMessageSend(const RawMessagePtr message) {
     LOG4CXX_ERROR(logger_, "Error while message deserialization.");
     return;
   }
+  std::map<uint8_t, uint32_t>::iterator it =
+      sessions_last_message_id_.find(sent_message.session_id());
 
-  if (((FRAME_TYPE_SINGLE == sent_message.frame_type()) ||
-       ((FRAME_TYPE_CONSECUTIVE == sent_message.frame_type()) &&
-        (0 == sent_message.frame_data())))) {
-    const auto last_message_it =
-        sessions_last_message_id_.find(sent_message.session_id());
-    if (sessions_last_message_id_.end() != last_message_it &&
-        sent_message.message_id() == last_message_it->second) {
-      sessions_last_message_id_.erase(last_message_it);
+  if (sessions_last_message_id_.end() != it) {
+    uint32_t last_message_id = it->second;
```

Comment:
@mvorobio const?

Reply:
@AByzhynar we decided to keep this code as is.